### PR TITLE
feat(rh-timestamp): add rh-timestamp

### LINKED ISSUE #548: [feat]: <rh-timestamp> ready to migrate
### Tag Name

pfe-timestamp

### RHDS Only

- [ ] The component will be **RHDS only**, delete the PFE component once published

### Additional Info

PFE Timestamp has already been updated in PF 1:1. We now need an `<rh-timestamp>` component for RHDS. No design tweaks necessary.

### PR DESCRIPTION
**Pull request by eyevana**
Closes #548

## What I did

1. Created a new element, `rh-timestamp`, based on `pf-timestamp`.
3. Fixed a type in the `feedback` partial that was returning the wrong markup for element pages.
4. Added docs

## Testing Instructions

1. Check out the demos and verify the examples are working as expected.
2. Verify docs match what Corey provided in #548.

### CODE REVIEW

File: elements/rh-timestamp/demo/rh-timestamp.html [hunk 1]
```diff
@@ -1,4 +1,90 @@
 <link rel="stylesheet" href="demo.css">
 <script type="module" src="rh-timestamp.js"></script>
 
-<rh-timestamp></rh-timestamp>
```
Comment (bennypowers) [inline elements/rh-timestamp/demo/rh-timestamp.html]: we'd prefer to split these out into separate demos. you can keep a `kitchen-sink.html` is that's easier for dev. see alert demos

**Review by bennypowers — CHANGES_REQUESTED** [review]

File: elements/rh-timestamp/demo/rh-timestamp.html (hunk 1 above)
Comment (eyevana) [reply elements/rh-timestamp/demo/rh-timestamp.html]: I see what you mean. Will do! :)

File: docs/cheatsheet.njk
Comment (bennypowers) [inline docs/cheatsheet.njk:1]: can we revert these unrelated changes and put them in a new PR? maybe you just need to rebase?

**Review by bennypowers — CHANGES_REQUESTED** [review]
needs a minor changeset. please copy an example from the changelog to maintain the format

**Comment by eyevana** [comment]
@coreyvickery timestamp docs are ready for review. Check it out :) https://deploy-preview-1149--red-hat-design-system.netlify.app/elements/

File: elements/rh-timestamp/README.md
Comment (eyevana) [inline elements/rh-timestamp/README.md:1]: @bennypowers do we still need this file? I'm asking because I noticed the latest element added to RHDS didn't have one.

Comment (zeroedin) [reply elements/rh-timestamp/README.md:1]: I believe we should still have this file even if with the components most basic usage and maybe a link to both docs and demo of the component.   As an example secondary nav's README is maybe overkill?

A more general reply and not specific to this PR alone is that we should ensure we have the same info in docs/demos/ and the README in sync such as component description / usage demo code.  Wondering if there is a better way to do this, maybe build these files automation pulling from the other sources we already maintain.

File: .changeset/strong-pens-boil.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+"@rhds/elements": minor
+---
+
+Added `<rh-timestamp>`.
+
+  Provides consistent formats for displaying date and time values.
+
+  ```html
+  <rh-timestamp></rh-timestamp>
```
Comment (bennypowers) [inline .changeset/strong-pens-boil.md:10]: This needs an attribute?

File: docs/scss/components/_example.scss [hunk 3]
```diff
@@ -66,6 +66,10 @@ rh-alert + .example {
     }
   }
 
+  &--component {
```
Comment (bennypowers) [inline docs/scss/components/_example.scss:69]: Can you do this with using sass concat or bem? We're trying to move away from those

File: elements/rh-timestamp/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Timestamp
+A timestamp provides consistent formats for displaying date and time values.
+
+## Usage
+Use a timestamp to display date and time values.
+
+```html
+<rh-timestamp></rh-timestamp>
```
Comment (bennypowers) [inline elements/rh-timestamp/README.md:8]: Attr?

Comment (bennypowers) [inline elements/rh-timestamp/README.md:8]: Attr and Gainesville content?

**Comment by eyevana** [comment]
@bennypowers thank you so much for the review! <3 I removed the BEM modifier (sorry, old habit), added some examples to the main demo file so users can preview in the playground, and added a specific date attribute to those examples you pointed out instead of defaulting to the current date. Let me know if you catch anything else.

File: .changeset/strong-pens-boil.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+---
+"@rhds/elements": minor
+---
+
+Added `<rh-timestamp>`.
+
+  Provides consistent formats for displaying date and time values.
+
+  ```html
+  <rh-timestamp date="Tue Aug 09 2006 14:57:00 GMT-0400"></rh-timestamp>
```
Comment (bennypowers) [inline .changeset/strong-pens-boil.md:10]:
```suggestion
  <rh-timestamp date="Tue Aug 09 2006 14:57:00 GMT-0400">Tue Aug 09 2006 14:57:00 GMT-0400</rh-timestamp>
```

File: elements/rh-timestamp/rh-timestamp.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  #timestamp = new TimestampController(this);
+
+  connectedCallback() {
+    super.connectedCallback();
+    if (this.hasAttribute('date')) {
+      this.#timestamp.date = new Date(this.getAttribute('date')!);
+    }
+  }
+
+  willUpdate(changedProperties: PropertyValues<this>) {
```
Comment (bennypowers) [inline elements/rh-timestamp/rh-timestamp.ts:98]:
```suggestion
  override willUpdate(changedProperties: PropertyValues<this>) {
```

File: elements/rh-timestamp/rh-timestamp.ts [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    return this.#timestamp.isoString;
+  }
+
+  get time() {
+    return this.#timestamp.time;
+  }
+
+  #timestamp = new TimestampController(this);
+
+  connectedCallback() {
```
Comment (bennypowers) [inline elements/rh-timestamp/rh-timestamp.ts:91]:
```suggestion
  override connectedCallback() {
```

File: elements/rh-timestamp/rh-timestamp.ts [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    }
+  }
+
+  willUpdate(changedProperties: PropertyValues<this>) {
+    for (const [prop] of changedProperties) {
+      this.#timestamp.set(prop, this[prop as keyof this]);
+    }
+  }
+
+  render() {
```
Comment (bennypowers) [inline elements/rh-timestamp/rh-timestamp.ts:104]:
```suggestion
  override render() {
```

**Review by bennypowers — APPROVED** [review]
last couple tweaks but lgtm

**Review by coreyvickery — APPROVED** [review]
